gnovm/pkg/gnolang/op_call.go

```diff
@@ -27,6 +27,11 @@ func (m *Machine) doOpPrecall() {
 	case TypeValue:
 		// Do not pop type yet.
 		// No need for frames.
+		xv := m.PeekValue(1)
+		if cx.GetAttribute(ATTR_SHIFT_RHS) == true {
+			xv.AssertNonNegative("runtime error: negative shift amount")
+		}
+
 		m.PushOp(OpConvert)
```

Comment:
Yeah that works. Could this be done the same way for shift assign operations by also modifying the RHS to type cast the value to uint?  If not, then my only remaining comment is to make the error message a constant because it is used in three places.

Reply:
shift assign is fixed, see:
https://github.com/gnolang/gno/blob/dd1c4161702115cb5e30ed60900591ab0f78af21/gnovm/tests/files/types/shift_e8.gno

https://github.com/gnolang/gno/blob/dd1c4161702115cb5e30ed60900591ab0f78af21/gnovm/tests/files/types/shift_e9.gno